Service: Random Number Generator (#217)
microsoft/jacdac: services/rng.md

# Random Number Generator

    identifier: 0x1789f0a2
    camel: rng

> A service that generates true random number generated by a HRND or TRNG.

Reviewer: ```suggestion
A service that generates random numbers using, where entropy is sourced from physical processes.
```

## Commands

    command random @ 0x80 {
        length: u8 { absolute_max=128 }
    }
    report random {
        data: bytes
    }

> A command that generates a random buffer with the given length.

Reviewer: ```suggestion
A command that generates a random buffer with the given length.
The response to this command is delayed when there isn't enough entropy available (similar to `/dev/random`).
```

Also, let's make this one optional, and add another one, I would call it `urandom` that doesn't wait for enough entropy, just uses a crypto-secure PRNG, seeded with real entropy.

    enum Variant: u32 {

>          TRNG = 1

Reviewer: ```suggestion
         Hardware = 1
```
